cpp/FLIP.h

```diff
@@ -2404,54 +2405,73 @@ namespace FLIP
-     * @param[in,out] parameters Contains parameters (e.g., PPD, exposure settings,etc). If the exposures have not been set by the user, then those will be computed (and returned).
-     * @param[in] imageWidth Width of the reference and test images.
-     * @param[in] imageHeight Height of the reference and test images.
      * @param[in] referenceThreeChannelImage Reference input image. For LDR, the content should be in [0,1]. The image is expected to have 3 floats per pixel and they
      *            are interleaved, i.e., they come in the order: R0G0B0, R1G1B1, etc. Input is expected to be in linear RGB.
      * @param[in] testThreeChannelImage Test input image. For LDR, the content should be in [0,1]. The image is expected to have 3 floats per pixel and they are interleaved.
-                  Input is expected to be in linear RGB
+                  Input is expected to be in linear RGB.
+     * @param[in] imageWidth Width of the reference and test images.
+     * @param[in] imageHeight Height of the reference and test images.
+     * @param[in,out] parameters Contains parameters (e.g., PPD, exposure settings,etc). If the exposures have not been set by the user, then those will be computed (and returned).
+     * @param[in] useHDR Set to true if the input images are to be considered containing HDR content, i.e., not necessarily in [0,1].
+     * @param[in] computeMeanFLIPError Set to true if the mean FLIP error should be computed. If false, mean error is set to 0.
      * @param[in] applyMagmaMapToOutput A boolean indicating whether the output should have the MagmaMap applied to it before the image is returned.
      * @param[out] errorMapFLIPOutput The computed FLIP error image is returned in this variable. If applyMagmaMapToOutput is true, the function will allocate
      *             three channels (and store the magma-mapped FLIP images in sRGB), and
-                   if it is false, only one channel will be allocated (and the FLIP error is returned in that grayscale image).
+     *             if it is false, only one channel will be allocated (and the FLIP error is returned in that grayscale image).
      *             Note that the user is responsible for deallocating the errorMapFLIPOutput image.
+     * @param[out] meanFLIPError Mean FLIP error in the test (testThreeChannelImage) compared to the reference (referenceThreeChannelImage).
      */
-    static void evaluate(const bool useHDR, FLIP::Parameters& parameters, const int imageWidth, const int imageHeight,
-        const float* referenceThreeChannelImage, const float* testThreeChannelImage, const bool applyMagmaMapToOutput, float** errorMapFLIPOutput)
+    static void evaluate(const float* referenceThreeChannelImage, const float* testThreeChannelImage,
+        const int imageWidth, const int imageHeight, FLIP::Parameters& parameters, const bool useHDR,
```

Comment:
Is there any argument for now having `parameters` before `useHDR`?
In all other cases, we have `useHDR` followed by `parameters`.

Reply:
Also, might look nice with this order?
`const bool applyMagmaMapToOutput, float** errorMapFLIPOutput, const bool computeMeanFLIPError, float& meanFLIPError,  `

Reply:
Fixed!